garage/tf/policies/categorical_conv_policy_with_model.py

```diff
@@ -0,0 +1,131 @@
+"""CategoricalConvPolicy with model."""
+from akro.tf import Discrete
+import tensorflow as tf
+
+from garage.misc.overrides import overrides
+from garage.tf.distributions import Categorical
+from garage.tf.models import CNNModel
+from garage.tf.models import MLPModel
+from garage.tf.policies.base2 import StochasticPolicy2
+
+
+class CategoricalConvPolicyWithModel(StochasticPolicy2):
+    """
+    CategoricalConvPolicy with model.
+
+    A policy that contains a CNN and a MLP to make prediction based on
+    a categorical distribution.
+
+    It only works with akro.tf.Discrete action space.
+
+    Args:
+        env_spec: Environment specification.
+        conv_filters: Number of filters.
+        conv_filter_sizes: Dimension of the filters.
+        conv_strides: The stride of the sliding window.
+        conv_pad: The type of padding algorithm to use,
+            either 'SAME' or 'VALID'.
+        name: Variable scope of the policy.
+        hidden_sizes: Output dimension of dense layer(s).
+        hidden_nonlinearity: Activation function for
+                    intermediate dense layer(s).
+        output_nonlinearity: Activation function for
+                    output dense layer.
+        layer_normalization: Bool for using layer normalization or not.
+
+    """
+
+    def __init__(self,
+                 env_spec,
+                 conv_filters,
+                 conv_filter_sizes,
+                 conv_strides,
+                 conv_pad,
+                 name='CategoricalConvPolicy',
+                 hidden_sizes=[],
+                 hidden_nonlinearity=tf.nn.relu,
+                 output_nonlinearity=tf.nn.softmax,
+                 layer_normalization=False):
+        assert isinstance(env_spec.action_space, Discrete), (
+            'CategoricalConvPolicy only works with akro.tf.Discrete'
+            'action space.')
+        super().__init__(name, env_spec)
+        self.obs_dim = env_spec.observation_space.flat_dim
+        self.action_dim = env_spec.action_space.n
+
+        self.add_model(
+            CNNModel(
+                filter_dims=conv_filter_sizes,
+                num_filters=conv_filters,
+                strides=conv_strides,
+                padding=conv_pad,
+                hidden_nonlinearity=hidden_nonlinearity,
```

Comment:
is `layer_normalization` only used in mlp? i checked that `batch_normalization` and `weight_normalization` also used in `garage.tf.core.network.ConvNetwork`.

Reply:
I saw that too and they are actually not used at all. I just removed them.